In [1]:
# Behzad Ghabaei 
# CS 82A - Data Science 
# Module 2 lab 1
# Python Warm - Up
# Instructor Seno
# 9/8/2026


### Step 1: Wages with f-strings

In [2]:
name = "Maria-Anna Chen"
hourly_wage = 16.00
hours = 40
weekly_pay = hourly_wage * hours
print(f"Maria-Anna earns {weekly_pay} per week") #640.0

Maria-Anna earns 640.0 per week


### Step 2: Count the warm days

In [3]:
temps = [61,65,72,58,70,74,68]
high = 0
for s in temps:
    if s > 65:
        high = high + 1
print(high) #4
        

4


### Step 3: Write a temperature converter

In [4]:
def to_celcius(f):
    return (f - 32) * 5/9

print(round(to_celcius(temps[0]), 1) ) #16.1

16.1


### Step 4: A movie dictionary

In [5]:
# {"key": "value"}
book = {"title": "Dune", "year": 1965, "rating": 100}
for key, value in book.items():
    print(key, ":", value)

title : Dune
year : 1965
rating : 100


### Step 5: Load and inspect chickwts.csv

In [6]:
import pandas as pd
df = pd.read_csv("chickwts.csv" ) # df = pd.read_csv("chickwts.csv", index_col = 0 )
df.head()
# df.tail()


,Unnamed: 0,weight,feed
0,1,179,horsebean
1,2,160,horsebean
2,3,136,horsebean
3,4,227,horsebean
4,5,217,horsebean


### df.shape would return (71, 3) with the code above written simply as: df = pd.read_csv("chickwts.csv"). This includes the Unnamed column.  When using the search box from "10 minutes to Pandas" and asking: "How to remove the Unnamed column?" The search returns, under "10 tools" the section called, "Index columns and trailing delimiters." Here is the snippet provided: 
```
In [94]: data = "index,a,b,c\n4,apple,bat,5.7\n8,orange,cow,10"

In [95]: pd.read_csv(StringIO(data), index_col=0)
Out[95]: 
            a    b     c
index                   
4       apple  bat   5.7
8      orange  cow  10.0
```
### When index_col = 0 is used in the parameters for data frame (df), such as: "df = pd.read_csv("chickwts.csv", index_col = 0 )" The command "df.shape" thus returns (71, 2).

In [7]:
df.shape

(71, 3)

In [8]:
df.describe()

,Unnamed: 0,weight
count,71.000000,71.000000
mean,36.000000,261.309859
std,20.639767,78.073700
min,1.000000,108.000000
25%,18.500000,204.500000
50%,36.000000,258.000000
75%,53.500000,323.500000
max,71.000000,423.000000


### Step 6: Interpret describe()

#### The **describe()** output tells us alot about the chick weight csv file. This is from the **Pandas** library. The **count** tells us how many chickens are participating in the survey.  The **mean** tells us the average weight in grams of all the 71 chickens combined.  The **std** is the standard deviation of all the chickens which describes the spread in between weights. A high std says the data is spread out.  The **min** is the lowest value in the data set in chicken weight (grams). The **max** is the highest weight in the dataset.  **10 minutes To Pandas** explains the describe() function, under "Viewing Data."  Describe shows a quick statistic summary of your data. This reference is called "pandas.DataFrame.describe" and gives the central tendancy, dispersion, and shape of dataset distribution, excluding NaN values. For numeric data, the results will include: count, mean, std, min, max, and lower percentile as 25, upper percentile as 75, and 50 percentile is the same as the median. ( pandas.pydata.org | https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.describe.html#pandas.DataFrame.describe )

#### **StackOverflow.com** explains, imagine sorting your data from lowest to highest.  **25%** means 25% of your data have the value 204.5 or below (less than or equal to).  **50%** means, 50% of your data have the value 258.0 or below, and 75% means, **75%** of you data have a value of 323.5 or below.  (My Prompt: "What does 25%, 50%, 75% mean in the describe() function?" | "What are 25%,50%,75% values when we describe a grouped dataframe?" | Sibbir Ahmed | https://stackoverflow.com/questions/57869926/what-are-25-50-75-values-when-we-describe-a-grouped-dataframe )

### End Of Lab - Behzad Ghabaei